test/models/term_fingerprint_test.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: term_fingerprints
+#
+#  id          :integer          not null, primary key
+#  fingerprint :string
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+require 'test_helper'
+
+class TermFingerprintTest < ActiveSupport::TestCase
+  test 'duplicate term fingerprints are not allowed' do
+    tf = TermFingerprint.first
+
+    assert_raises(ActiveRecord::RecordInvalid) do
+      TermFingerprint.create!(fingerprint: tf.fingerprint)
+    end
+  end
+
+  test 'deleting a TermFingerprint does not delete its Term, which is still valid' do
+    term_count = Term.count
+    fingerprint_count = TermFingerprint.count
+
+    target = TermFingerprint.last
+    target_term = target.terms.last
+
+    assert_operator 0, :<, target.terms.count
+
+    target.destroy
+    target_term.reload
+
+    assert_equal term_count, Term.count
+    assert_equal fingerprint_count - 1, TermFingerprint.count
+    assert_instance_of NilClass, target_term.term_fingerprint
```

Comment:
You may want to consider `assert_nil` here:
https://ruby-doc.org/3.0.5/gems/minitest/Minitest/Assertions.html#method-i-assert_nil

Both work, but I think the more explicit assertion is slightly more efficient to mentally process (for me at least)